pkg/server/metrics.go

```diff
@@ -19,31 +19,32 @@ var (
 		Name: "handled_events_total",
 		Help: "Total number of handled events.",
 	}, []string{"event_type"})
+	ghClient ghclient.Client
 )
 
 // RegisterMetrics registers prometheus collectors to collect metrics
-func RegisterMetrics(client ghclient.Client) (*Metrics, []error) {
+func RegisterMetrics(client ghclient.Client) ([]error) {
 	errors := make([]error, 0, 3)
-	metrics := &Metrics{
-		ghClient: client,
-	}
-
-	registerOrGet(rateLimit, &errors, func(collector prometheus.Collector) {
-		metrics.RateLimit = collector.(*prometheus.GaugeVec)
+	ghClient = client
+	RegisterOrGet(rateLimit, &errors, func(collector prometheus.Collector) {
+		rateLimit = collector.(*prometheus.GaugeVec)
 	})
 
-	registerOrGet(webHookCounter, &errors, func(collector prometheus.Collector) {
-		metrics.WebHookCounter = collector.(*prometheus.CounterVec)
+	RegisterOrGet(webHookCounter, &errors, func(collector prometheus.Collector) {
+		webHookCounter = collector.(*prometheus.CounterVec)
 	})
 
-	registerOrGet(handledEventsCounter, &errors, func(collector prometheus.Collector) {
-		metrics.HandledEventsCounter = collector.(*prometheus.CounterVec)
+	RegisterOrGet(handledEventsCounter, &errors, func(collector prometheus.Collector) {
+		handledEventsCounter = collector.(*prometheus.CounterVec)
 	})
 
-	return metrics, errors
+	return errors
 }
 
-func registerOrGet(collector prometheus.Collector, errors *[]error, assign func(regCollector prometheus.Collector)) {
+// RegisterOrGet registers the provided Collector with the DefaultRegisterer and
+// assigns the Collector, unless an equal Collector was registered before, in
+// which case that Collector is assigned.
+func RegisterOrGet(collector prometheus.Collector, errors *[]error, assign func(regCollector prometheus.Collector)) {
```

Comment:
just please remind me why the method is named `RegisterOrGet` and not `RegisterMetric` - I'm not sure if we have talked about it or not...

Reply:
I think I forgot to rename it after lots of refactoring for first metrics pr. Renaming to `RegisterOrAssignCollector`